Overall changes in the app hieracrchical structure and new features

MaorBezalel opened the pull request:
# Changes Made

- Improved the app's visual appearance for an enhanced user experience.
- Changed CSS classes naming to follow the BEM (Block, Element, Modifier) convention ([BEM Documentation](https://getbem.com/naming/)).
- Renamed some of the components for consistency and clarity:
  - Renamed Navigation to RouteEditor.
  - Renamed RouteComp to Route.
  - Renamed StepComp to Step.

- Removed the `MenuComp` component, as its functionality has been integrated into the `RouteEditor` component.

- Changed Hierarchical Structure:
  - Modified the structure as follows:
    ```
    App
    └── RouteManager - Contains the Route Editor and Tutorial components (2 separate modes).
        └── RouteEditor - The main component where the user can manage their routes and steps.
            └── Route - A component that represents a route.
                └── Step - A component that represents a step.
        └── RouteTutorial - A component that explains how to use the RouteEditor.
    PopupWindow - A general component that opens a popup window with a title, main content, and buttons.
    ```
  - Reorganized components to improve clarity and maintainability.
  - Adjusted component nesting for better modularity.

- Fixed a typo in one of our folders - `style` to `styles`.

# New Features
- Added the `RouteManager` component, which aims to ease the transition between RouteEditor and RouteTutorial (TODO).
- Added the `PopupWindow` component, which serves as a general component outside the app and can be used when we want to receive immediate input from the user.
- Added the ability to **remove routes** (via `button onClick` event).
- Added the ability to also **remove steps** of a specific route (via `button onClick` event).
- Added automatic numbering to routes and steps, updating the numbering upon deletion.

# Incomplete Tasks

While I've made significant progress, there are still some pending tasks that I haven't been able to complete:

- Couldn't figure out how to expand/collapse the routes to reveal/hide their steps.
- Haven't implemented the functionality behind the `Select Mode` buttons, which are supposed to allow the user to select any routes/steps they wish, which can then enable deleting them by pressing the `Delete\Remove` buttons.
- Haven't implemented the functionality behind the `Select All` buttons, which are supposed to auto-select all routes/steps, and then allow the user to delete them by pressing the `Delete\Remove` buttons.

# Ideas for the Future:
- When a user tries to delete/remove any route or step they created, it might be a good idea to use the `PopupWindow` component to ask if it is indeed what they intend to do, and to warn them that any data deleted can't be brought back.
- We should add some colors to the buttons to make it easier to distinguish between them (<span style="color: green;">Green</span> for `Add` buttons... <span style="color: red;">Red</span> for `Delete\Remove` buttons... etc.').

I'm looking forward to feedback and suggestions on the changes I've made so far. Please let me know if there are any further improvements needed or if you have any questions.

![‪Navigation - Google Chrome‬ 17_08_2023 23_05_02](https://github.com/NadavBitran/AR-Routing-UI/assets/122044231/92801838-983b-480f-8af1-d17c2627e836)
![‪Navigation - Google Chrome‬ 17_08_2023 23_05_24](https://github.com/NadavBitran/AR-Routing-UI/assets/122044231/c718d203-5f2c-4c46-9f7a-f62a130e4007)
![‪Navigation - Google Chrome‬ 17_08_2023 23_05_31](https://github.com/NadavBitran/AR-Routing-UI/assets/122044231/0be6bf9a-000c-43a9-b921-27dfcee1547b)
![‪Navigation - Google Chrome‬ 17_08_2023 23_05_37](https://github.com/NadavBitran/AR-Routing-UI/assets/122044231/6bd78d2a-8bb2-46da-99a9-3b3e72b149e9)
![‪Navigation - Google Chrome‬ 17_08_2023 23_05_43](https://github.com/NadavBitran/AR-Routing-UI/assets/122044231/29de0c3f-f3ec-4a29-97a7-32b429bdb508)

MaorBezalel committed:
Overall change in the app hieracrchical structure and new features

issac211 commented:
Really nice work!!
but it looks like you're overriding the changes I made.

We need to go over it before doing a merge..

MaorBezalel committed:
Fixed the issue that overrode some of the previous merge changes (typos fix & button for printing out the routeList)

MaorBezalel committed:
Placed the routeList in RouteManager instead of RouteEditor so that
in the future we will be able to pass from the RouteManager  the routeList as JSON to next stage in the pipeline.

MaorBezalel commented:
> Really nice work, but it looks like you're overriding the changes I made. Is there a chance you did a pull and then copied your code? It's not good, because there should have been a conflict
> 
> Conflicts are important That way we know where our changes collide Now if we do a merge, it will overwrite what I did :(
> 
> We need to go over it before doing a merge

Hi Isaac,

Thank you for bringing up the issue regarding the override problem. I've reviewed the changes and indeed noticed that some of your typo fixes and the `print button` which prints the `routeList` for debugging purposes from the last merge were indeed missing in my branch. I made the necessary fixes to ensure that your changes are now included in my branch.

Here's a summary of the changes I addressed:

- Incorporated the typos fix made by you (`routeid` to `routeId`; `stepid` to `stepId`).
- Added a print button to output the `routeList` to the console for debugging purposes. I also made the print output the `routeList` in a more organized way.
- Ensured the creation of the `routeList` is in the `RouteManager` component, making it easier for us in the future to pass the `routeList` as JSON to the next stage in the pipeline.

I apologize for any confusion caused. Please feel free to review the updated changes in the pull request, and if you have any further feedback or suggestions, please let me know.

Best regards,  
Maor Bezalel.

MaorBezalel committed:
Aded new features

issac211 committed:
Fix: conflicts
